axelor-stock/src/main/java/com/axelor/apps/stock/web/MassStockMoveController.java

```diff
@@ -0,0 +1,129 @@
+package com.axelor.apps.stock.web;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.service.exception.TraceBackService;
+import com.axelor.apps.stock.db.MassStockMove;
+import com.axelor.apps.stock.db.PickedProducts;
+import com.axelor.apps.stock.db.StoredProducts;
+import com.axelor.apps.stock.db.repo.MassStockMoveRepository;
+import com.axelor.apps.stock.exception.StockExceptionMessage;
+import com.axelor.apps.stock.service.MassStockMoveService;
+import com.axelor.apps.stock.service.PickedProductsService;
+import com.axelor.apps.stock.service.StoredProductsService;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+
+public class MassStockMoveController {
+
+  public void importProductFromStockLocation(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    MassStockMove massStockMove = request.getContext().asType(MassStockMove.class);
+    massStockMove = Beans.get(MassStockMoveRepository.class).find(massStockMove.getId());
+    Beans.get(MassStockMoveService.class).importProductFromStockLocation(massStockMove);
+    response.setReload(true);
+  }
+
+  @Transactional
+  public void realizePicking(ActionRequest request, ActionResponse response) {
+    MassStockMove massStockMove = request.getContext().asType(MassStockMove.class);
+    massStockMove = Beans.get(MassStockMoveRepository.class).find(massStockMove.getId());
+    boolean isQtyEqualZero = false;
+    boolean isGreaterThan = false;
+    PickedProductsService pickedProductsService = Beans.get(PickedProductsService.class);
+    for (PickedProducts pickedProducts : massStockMove.getPickedProductsList()) {
```

Comment:
I think the `for` boucle should be in a service.

From what I understood, relalizePicking must not stop at a exception, which might be the reason why TraceBackService.trace does not use response, therefore there is no reason to do it in a service.
Since it seems we need to have alert, we can just do a check before calling our need method and set alert before